openedx/core/djangoapps/user_authn/views/login.py

```diff
@@ -576,13 +576,26 @@ def login_user(request, api_version='v1'):
             if possibly_authenticated_user and password_policy_compliance.should_enforce_compliance_on_login():
                 # Important: This call must be made AFTER the user was successfully authenticated.
                 _enforce_password_policy_compliance(request, possibly_authenticated_user)
-                check_pwned_password_and_send_track_event.delay(user.id, request.POST.get('password'), user.is_staff)
 
         if possibly_authenticated_user is None or not (
             possibly_authenticated_user.is_active or settings.MARKETING_EMAILS_OPT_IN
         ):
             _handle_failed_authentication(user, possibly_authenticated_user)
 
+        pwned_properties = check_vulnerable_password(
+            user.id, request.POST.get('password'), user.is_staff
+        ) if not is_user_third_party_authenticated else {}
+        password_frequency = pwned_properties.get('frequency', 0)
+        if (
+            settings.ENABLE_AUTHN_LOGIN_HIBP_BLOCK and
+            password_frequency >= settings.LOGIN_HIBP_BLOCK_FREQUENCY_THRESHOLD
```

Comment:
```suggestion
            settings.ENABLE_AUTHN_LOGIN_HIBP_BLOCK and
            password_frequency >= settings.LOGIN_HIBP_BLOCK_FREQUENCY_THRESHOLD and
            not is_user_third_party_authenticated
```

Reply:
we don't need to add another condition here. We are already giving a default value to pwned_properties here:
https://github.com/openedx/edx-platform/blob/a3eca17db678723487203c043171d561aa841923/openedx/core/djangoapps/user_authn/views/login.py#L585-L588

and password_frequency will be 0 (zero). 
```python
password_frequency >= settings.LOGIN_HIBP_BLOCK_FREQUENCY_THRESHOLD
# 0 >= settings.LOGIN_HIBP_BLOCK_FREQUENCY_THRESHOLD will return False
```